api/src/certification/course/infrastructure/repositories/sco-certification-candidate-repository.js

```diff
@@ -4,22 +4,28 @@ import { knex } from '../../../../../db/knex-database-connection.js';
  * @returns {Array<number>} certification candidates ids
  */
 const findIdsByOrganizationIdAndDivision = async function ({ organizationId, division }) {
-  return knex
-    .select(['certification-candidates.id'])
-    .from('certification-candidates')
-    .join(
-      'view-active-organization-learners',
-      'view-active-organization-learners.id',
-      'certification-candidates.organizationLearnerId',
+  const subquery = knex
+    .select(
+      'certification-candidates.id',
+      knex.raw(
+        'row_number() OVER (PARTITION BY "certification-candidates"."organizationLearnerId" ORDER BY "sessions"."publishedAt" DESC NULLS LAST) AS row_number',
+      ),
     )
+    .from('view-active-organization-learners as learners')
+    .innerJoin('certification-candidates', 'learners.id', 'certification-candidates.organizationLearnerId')
+    .innerJoin('sessions', 'certification-candidates.sessionId', 'sessions.id')
     .where({
-      'view-active-organization-learners.organizationId': organizationId,
-      'view-active-organization-learners.isDisabled': false,
+      'learners.organizationId': organizationId,
+      'learners.isDisabled': false,
     })
-    .whereRaw('LOWER("view-active-organization-learners"."division") = ?', division.toLowerCase())
+    .whereRaw('LOWER(learners.division) = ?', division.toLowerCase())
     .orderBy('certification-candidates.lastName', 'ASC')
-    .orderBy('certification-candidates.firstName', 'ASC')
-    .pluck('certification-candidates.id');
+    .orderBy('certification-candidates.firstName', 'ASC');
+
+  return knex
+    .pluck('id')
+    .from(subquery.as('uniqLastCandidatesByOrganizationLearners'))
```

Comment:
en relisant ça avec stephen, on se dit que nommer la constante subquery sera plus lisible:

```suggestion
     .from(uniqLastCandidatesByOrganizationLearners)
```